src/platform/linux/wayland/window.rs

```diff
@@ -273,9 +268,23 @@ impl Window {
     }
 
     #[inline]
-    pub fn set_cursor_position(&self, _pos: LogicalPosition) -> Result<(), ()> {
-        // TODO: not yet possible on wayland
-        Err(())
+    pub fn set_cursor(&self, _cursor: MouseCursor) {
+        unimplemented!("Set cursor is not yet possible on Wayland.")
+    }
+
+    #[inline]
+    pub fn hide_cursor(&self, _hide: bool) {
+        unimplemented!("Cursor hiding is not yet possible on Wayland.")
+    }
+
+    #[inline]
+    pub fn grab_cursor(&self, _grab: bool) -> Result<(), String> {
+        unimplemented!("Cursor grabbing is not yet possible on Wayland.")
```

Comment:
I think using `Err` is probably better here, since `unimplemented!` is used more as a call to action; from what I understand, it's not possible for us to implement this on Wayland yet even if we wanted to, so panicking here could make writing multi-plat apps really cumbersome. The same applies to `set_cursor_position`.

Since `hide_cursor` doesn't return a `Result`, you don't have that option there... it's better to fail silently then to force users to manually check if the Wayland backend is in use before calling the method. Once logging is added, it can just emit a warning, which should be better than nothing (though, this sort of stuff should really be in the docs too, but I don't expect you to do that).

I'm under the impression that `set_cursor` is actually possible, though @vberger would know that better than I would. In that case, my only nitpick is on the use of "yet" and it being "Set cursor" instead of "Setting the cursor".

Reply:
The 'yet' part of the error message pre-date the changes I've made.